src/expr/src/scalar/mod.rs

```diff
@@ -1899,6 +1899,40 @@ impl MirScalarExpr {
         })?;
         Ok(size)
     }
+
+    /// Extract all dependencies of `self`.
+    pub fn depends_on(&self) -> BTreeSet<GlobalId> {
+        let mut dependencies = BTreeSet::new();
```

Comment:
This variable isn't ever extended with anything other than a call to `depends_on`, which means it's always empty. I believe this function will need to be passed a Scope or something with which it can generate a GlobalId from a Column index.

Reply:
Wait no, MirScalarExprs can never depend on GlobalIds because only MirRelationExprs do that.

Reply:
Added a commit that cleans this up, should be no logical change though.